zebra-network/src/address_book.rs

```diff
@@ -72,6 +73,10 @@ pub struct AddressBook {
     /// [`OrderedMap`] sorts in descending order.
     by_addr: OrderedMap<PeerSocketAddr, MetaAddr, Reverse<MetaAddr>>,
 
+    /// The address with a last_connection_state of [`PeerAddrState::Responded`] and
+    /// the most recent `last_response` time by IP.
```

Comment:
> It's hard to say though, because it's only really important for testing.

I could open a suggestion PR replacing `has_active_peer_with_ip()` with `num_active_peers_with_ip()` to make the review a little easier if you think it may be worth doing for the tests?

> What do you think users would want?

I think users would expect `max_conns_per_ip` to at least configure the number of connections Zebra will accept, but that it would be better for it to define how many connections Zebra will try to initiate as well.

Reply:
> > It's hard to say though, because it's only really important for testing.
> 
> I could open a suggestion PR replacing `has_active_peer_with_ip()` with `num_active_peers_with_ip()` to make the review a little easier if you think it may be worth doing for the tests?

That's a different change with a lower priority, because it's not a security fix. Also, I'd prefer we all shift to focusing on the RPC work soon, rather than adding more network changes. (I'd like to get our current set of changes tested in a release first.)

But ultimately this is @mpguerra's decision, and she uses tickets to make those decisions. 

> > What do you think users would want?
> 
> I think users would expect `max_conns_per_ip` to at least configure the number of connections Zebra will accept, but that it would be better for it to define how many connections Zebra will try to initiate as well. It's not worth doing immediately unless we need it for integration tests.

I think you're right, and it's ok as long as we document it.

Reply:
Documented in https://github.com/ZcashFoundation/zebra/pull/7029/commits/d57f2b1c54883ae56ecec8f7fafba21a43663bc9.

Reply:
I was thinking about this a bit more yesterday. Some users will need Zebra to accept multiple connections from the same IP, either for testing, or for internal caching setups.

So could we disable the checks in this PR when `max_connections_per_ip > 1`?
(And document the security implications on that config.)

That way, the default config is secure, but users who need it can get Zebra to make multiple outbound connections.

Reply:
Great idea, thank you, applied in https://github.com/ZcashFoundation/zebra/pull/7029/commits/6d17a9cbd0e37f2a5196d046e2c94dfc0eacd043.